chips/sam4l/src/adc.rs

```diff
@@ -408,12 +447,19 @@ impl hil::adc::Adc for Adc {
             self.active.set(true);
             self.continuous.set(true);
 
+            let trgsel;
+            if self.cpu_clock.get() {
+                trgsel = 1; // internal timer trigger
```

Comment:
Hmm, that does match my experiences. The ADC seems quite picky about input clock speeds. As long as it keeps functioning steadily in continuous mode though, that seems fine.